draft-cpbs-pseudorandom-ctls.md

```diff
@@ -172,6 +172,16 @@ CTLSPlaintext records are subject to an additional decipherment step:
 2. Let `tweak` be `"client datagram hs" + profile_id + Handshake.msg_type` if sent by the client, or `"server datagram hs" + profile_id + Handshake.msg_type` if sent by the server.
 3. Replace `Handshake.body` with `STPRP-Decipher(key, tweak, Handshake.body)`.
 
+## Optional Keys
+
+### "start-tag"
+
+In some deployments, it may be necessary or valuable to authenticate that a ClientHello was issued with a particular Pseudorandom cTLS configuration.  Merely deciphering the ClientHello, and observing that it parses correctly, does not conclusively prove that it was issued as expected.  It is possible, though not likely, that a random byte sequence could be parsed as a valid cTLS ClientHello.
+
+To require proof that the sender was using a particular Pseudorandom cTLS template, the template MAY include the optional key "start-tag" whose value is an integer `T` in the range `1..32`.  When this key is present, the client MUST prepend `T` zero bytes to the ClientHello before calling `STPRP-Encipher()` on the ClientHello payload, and the server SHOULD verify that these bytes are zero after calling `STPRP-Decipher()`.  This arrangement authenticates the STPRP.  The indicated length in the `Handshake` header is not altered, so this does not reduce the maximum handshake message size.
+
+Start tags can be used to enable rotation of the Pseudorandom cTLS key.  With a tag in place, servers can use trial decryption to identify which key (old or new) is in use for each new connection.  To avoid false matches, values of `T` less than 4 are NOT RECOMMENDED.
```

Comment:
```suggestion
Start tags can be used to enable rotation of the Pseudorandom cTLS key.  With a tag in place, servers can use trial decryption to identify which key (old or new) is in use for each new connection.  To avoid false matches, values of `T` less than 8 are NOT RECOMMENDED.
```